app/models/test_passage.rb

```diff
@@ -0,0 +1,62 @@
+class TestPassage < ApplicationRecord
+
+  SUCCESS_PERCENT = 85
+
+  belongs_to :user
+  belongs_to :test
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :before_validation_set_first_question, on: :create
+  before_validation :before_validation_set_next_question, on: :update
+
+  def completed?
+    current_question.nil?
+  end
+
+  def accept!(answer_ids)
+    if correct_answer?(answer_ids)
+      self.correct_questions += 1
+    end
+
+
+    save!
+  end
+
+  def current_question_number
+    test.questions.order(:id).where('id <= ?', current_question.id).count
+  end
+
+  def success_rate
+    (correct_questions * 100) / test.questions.count
+  end
+
+  def successful?
+    success_rate >= SUCCESS_PERCENT
+  end
+
+  private
+
+  def before_validation_set_first_question
+    self.current_question = test.questions.first if test.present?
+  end
+
+  def before_validation_set_next_question
+    self.current_question = next_question if new_record? == false
+  end
+
+  def correct_answer?(answer_ids)
+    correct_answers_count = correct_answers.count
+
+    (correct_answers_count == correct_answers.where(id: answer_ids).count) &&
+    correct_answers_count == answer_ids.count
+  end
+
+  def correct_answers
+    current_question.answers.correct
+  end
+
+  def next_question
+    test.questions.order(:id).where('id > ?', current_question.id).first
```

Comment:
Same thing, if you want to find question without answer for this test, you should find all questions without answer and then sort them and get 1